tsl/src/compression/compress_utils.c

```diff
@@ -413,10 +413,12 @@ decompress_chunk_impl(Oid uncompressed_hypertable_relid, Oid uncompressed_chunk_
 }
 
 bool
-tsl_compress_chunk_wrapper(Chunk *chunk, bool if_not_compressed)
+tsl_compress_chunk_wrapper(Chunk *chunk, bool if_not_compressed, bool ignore_errors)
```

Comment:
Would suggest something like `idempotent`, which we can use for other functions that we want to behave in an idempotent manner even if it has nothing to do with errors.

Reply:
I see what you're saying, but `if_not_compressed` is already the idempotent option. It's already kind of confusing. Ideally, we should merge those 2 arguments, but I kept `if_not_compressed` for backward-compatibility reasons.

Reply:
what is the difference between the 2 arguments, why do we need the one you are introducing?

Reply:
The relevant documentation will be [here](https://github.com/timescale/docs.timescale.com-content/pull/706/files).

The difference is that with this argument set to `true` then remote execution of `compress_chunk()` is guaranteed to return consistent results between all Data Nodes, even if the replicas are not identical with respect to their compression status (e.g. one replica is compressed, the other replica is not).

Originally I changed the semantics of `if_not_compressed` to suit my needs because the return value of `compress_chunk()` was undocumented. However, the risk of breaking existing code that may be using `compress_chunk()` in this undocumented manner made me reconsider and take @mkindahl 's suggestion to introduce a third "silent" flag.

Reply:
The team has decided to change the return value of this function and remove the extra argument. We still need to add this to documentation.